File: DSA PRACTICE/LECTURE8/NumberOfBit.cpp
//Total number of set bit in a
#include<iostream>
using namespace std;

void bit(int a)
{
    int abit=0;
	while(a>0)
	{
		if(a&1)
		{
			abit=abit+1;
		}
		a=a>>1;
	}
    cout<<abit;
	
}

int main()
{
	int a;
	cin>>a;

	bit(a);
    return 0;
}


/*
INPUT:
a=2 -> 01

OUTPUT:
Total 1's= 1
*/ 

File: DSA PRACTICE/LECTURE8/NumberOfBit.java
//Total number of set bit in a&b
import java.util.Scanner;
public class NumberOfBit
{
	public static void main(String[] args) 
	{
		Scanner sc=new Scanner(System.in);
		System.out.println("Eneter the number a and b");
		int a=sc.nextInt(), b=sc.nextInt();

		bit(a,b);
	}

	public static void bit(int a,int b)
	{
	    int aBit=0, bBit=0;
		while(a>0 || b>0)
		{
			int aBit1=a&1, bBit1=b&1;
			if(aBit1==1)
			{
				aBit=aBit+1;
			}
			if(bBit1==1)
			{
				bBit=bBit+1;
			}
			a=a>>1;
			b=b>>1;
		}
		System.out.print("Total 1's in a and b: "+(aBit+bBit));
	}
}


/*
INPUT:
a=2 -> 01
b=3 -> 11

OUTPUT:
Total 1's= 3

.................
Eneter the number a and b
2
3
Total 1's in a and b: 3
*/ 